src/include/sof/audio/audio_stream.h

```diff
@@ -228,6 +228,8 @@ static inline void *audio_stream_wrap(const struct audio_stream *buffer,
 		ptr = (char *)buffer->addr +
 			((char *)ptr - (char *)buffer->end_addr);
 
+	assert((intptr_t)ptr <= (intptr_t)buffer->end_addr);
```

Comment:
Lets log this as an error instead as it unrecoverable if deployed in production.

Reply:
ping - we need to be more recoverable when we fail. e.g. so we can attempt graceful powerdown and reboot.

Reply:
I know assert() is in other places, but the intention is to remove as we go along - we should only really be asserting if its likely impossible to gracefully recover.